### PR TITLE
feat(sig): add scalability group

### PR DESCRIPTION
**Pull request by morey-tech**
Create the Scalability SIG based on the discussion in the community meeting on 2023-03-01 about [`docs: add scalability benchmarking proposal #12662`](https://github.com/argoproj/argo-cd/pull/12662).

Todo, pending approval of this PR:
- [x] Create the `argo-sig-scalability` channel on CNCF Slack.
- [x] Add the SIG Scalability meetings to the argoproj calendar.
- [x] Create a Google Doc for the meetings and add the argoproj google account as the owner.
- [x] Add GitHub App to Slack channel to post notifications for [the `type:scalability` label](https://github.com/argoproj/argo-cd/pulls?q=is%3Apr+is%3Aopen+label%3Atype%3Ascalability).

### CODE REVIEW

File: sigs/sig-scalability/README.md
Comment (morey-tech) [inline sigs/sig-scalability/README.md:3]: Given that the proposal leading the creation of this SIG is focused on Argo CD, I've included that in the description. If the scope changes in the future, we can update the SIG.

Though this leads me to consider calling this the `Argo CD SIG Scalability`, I'm unsure if that is correct.

File: sigs/sig-scalability/README.md
Comment (morey-tech) [inline sigs/sig-scalability/README.md:10]: Another way to contact the SIG could be using [the `type:scalability` label](https://github.com/argoproj/argo-cd/pulls?q=is%3Apr+is%3Aopen+label%3Atype%3Ascalability) on PRs and issues.

File: sigs/sig-scalability/README.md
Comment (csantanapr) [reply sigs/sig-scalability/README.md:3]: I think we can call it "Argo SIG Scalability" and just state initial focus of the SIG is ArgoCD only and the scope described here. If there is a case that there are folks that want to join and focus on something else like Workflows, they can go ahead do a proposal to expand the SIG scope but no need for a new SIG.

File: sigs/sig-scalability/README.md
Comment (csantanapr) [reply sigs/sig-scalability/README.md:8]: I think this is fine

File: sigs/sig-scalability/README.md
Comment (nimakaviani) [reply sigs/sig-scalability/README.md:3]: i totally agree with @csantanapr 's suggestion. ideally this should grow beyond CD and we can step up to help with other argo projects as well.

File: sigs/sig-scalability/README.md
Comment (jannfis) [inline sigs/sig-scalability/README.md:19]:
```suggestion
| [Andrew Lee](https://github.com/andklee)        | AWS     |       |
| [Carlos Santana](https://github.com/csantanapr) | AWS     |       |
| [Jann Fischer](https://github.com/jannfis)      | Red Hat |       |
| [Nicholas Morey](https://github.com/morey-tech) | Akuity  |       |
| [Nima Kaviani](https://github.com/nimakaviani)  | AWS     |       |
```

File: sigs/sig-scalability/README.md
Comment (waltforme) [inline sigs/sig-scalability/README.md:17]: Jun Duan, https://github.com/waltforme, IBM

File: sigs/sig-scalability/README.md
Comment (morey-tech) [reply sigs/sig-scalability/README.md]: [Done](https://github.com/argoproj/argoproj/pull/192/commits/616cb1583dc886881d6ff8c029c5d869c64ce4d7)!

**Comment by hblixt** [comment]
To create a SIG, you need a maintainer appointed secretary/coordinator, (https://github.com/argoproj/argoproj/blob/master/community/SIG-GOVERNANCE.md) who also needs to be an Argo maintainer. That need to be figured out and the person appointed by the Argo maintainers before this PR can be merged and the SIG created
As there are also plans for an argo-labs repos, a project onboarding ticket is needed (https://github.com/argoproj/argoproj/blob/master/proposals/001-project-onboarding.md)
It might also be worth considering not intertwining the SIG and the repos too much e.g. the SIG has a broader charter around Argo scalability, whereas the argo-labs tools repos is focused on implementation but has an ownership/membership that's different than the SIG itself.

File: sigs/sig-scalability/README.md
Comment (hblixt) [inline sigs/sig-scalability/README.md:5]: I think this can be handled within the SIG without the need for a new separate proposal, by giving the SIG a bit more loose charter. Doesnt mean that other areas will be covered now (or ever) but reduces some process.

File: sigs/sig-scalability/README.md
Comment (hblixt) [inline sigs/sig-scalability/README.md:15]: I assume this will have open membership, e.g. 
"Membership is open to any interested person in the Argo community"

**Comment by morey-tech** [comment]
Thank you, @hblixt, for your review and guidance!

> To create a SIG, you need a maintainer appointed secretary/coordinator, (https://github.com/argoproj/argoproj/blob/master/community/SIG-GOVERNANCE.md). That needs to be figured out, and the person appointed by the Argo maintainers before this PR can be merged and the SIG created

I'm happy to take on the coordinator role for the Scalability SIG. What is the appointment process, or where is best to reach out for an approval from the maintainers? It would be good to add it to the SIG README, I could add a `Role` column in the Members table.

> As there are also plans for an argo-labs repos, a project onboarding ticket is needed (https://github.com/argoproj/argoproj/blob/master/proposals/001-project-onboarding.md) It might also be worth considering not intertwining the SIG and the repos too much e.g. the SIG has a broader charter around Argo scalability, whereas the argo-labs tools repos is focused on implementation but has an ownership/membership that's different than the SIG itself.

Understood. The repo for the scalability benchmarking tooling will **not** be limited to the Scalability SIG and vice versa.

**Comment by hblixt** [comment]
> Thank you, @hblixt, for your review and guidance!
> 
> 
> I'm happy to take on the coordinator role for the Scalability SIG. What is the appointment process, or where is best to reach out for an approval from the Maintainers? It would be good to add it to the SIG README, I could add a `Role` column in the Members table.
> 

I have added it as a topic to the Argo Maintainer meeting on 3/7. Would be good if you could be there @morey-tech

**Comment by morey-tech** [comment]
> I have added it as a topic to the Argo Maintainer meeting on 3/7. Would be good if you could be there @morey-tech

Great, thanks! I'll be there.

File: sigs/sig-scalability/README.md
Comment (todaywasawesome) [inline sigs/sig-scalability/README.md:23]:
```suggestion
| [Carlos Santana](https://github.com/csantanapr) | AWS     |       |
| [Dan Garfield](https://github.com/todaywasawesome) | Codefresh     |       |
```

I'd like to add myself here as I've done some work on the gen resources tooling at this point.

**Comment by morey-tech** [comment]
Based on the conversation in the [Argo Maintainers Meeting on 2023-03-07](https://docs.google.com/document/d/1C71GA0XCxsyGP7Q4SYveHak5ZZcJOuaOiDCt_xHG10U/edit#heading=h.moq3auks0mh4), we will move forward with holding the SIG Scalability meetings to promote engagement from the broader community before proceeding with the formal approval.

In support of this, I'll arrange to create the meeting log document, Slack channel, and schedule the recurring meeting on the argoproj calendar.

File: sigs/sig-scalability/README.md
Comment (csantanapr) [reply sigs/sig-scalability/README.md]: Cool @daniel-codefresh 🙌

**Comment by csantanapr** [comment]
@morey-tech 
>In support of this, I'll arrange to create the meeting log document, Slack channel, and schedule the recurring meeting on the argoproj calendar.

your a star ⭐

File: sigs/sig-scalability/README.md
Comment (crenshaw-dev) [inline sigs/sig-scalability/README.md:28]:
```suggestion
| [Andrew Lee](https://github.com/andklee)            | AWS       |       |
| [Carlos Santana](https://github.com/csantanapr)     | AWS       |       |
| [Dan Garfield](https://github.com/todaywasawesome)  | Codefresh |       |
| [Jann Fischer](https://github.com/jannfis)          | Red Hat   |       |
| [Jun Duan](https://github.com/waltforme)            | IBM       |       |
| [Michael Crenshaw](https://github.com/crenshaw-dev) | Intuit    |       |
| [Nicholas Morey](https://github.com/morey-tech)     | Akuity    |       |
| [Nima Kaviani](https://github.com/nimakaviani)      | AWS       |       |
| [Zach Aller](https://github.com/zachaller)          | Intuit    |       |
```

File: sigs/sig-scalability/README.md
Comment (leoluz) [inline sigs/sig-scalability/README.md:26]: Hi!
Can you please add me to this group as well?
`| [Leonardo Luz Almeida](https://github.com/leoluz) | Intuit    |       |`

File: sigs/sig-scalability/README.md
Comment (vsethi) [inline sigs/sig-scalability/README.md:31]: Please add me as a member. Thanks!
```suggestion
| [Vikram Sethi](https://github.com/vsethi)          | Adobe    |       |
| [Zach Aller](https://github.com/zachaller)          | Intuit    |       |
```

**Review by vsethi — COMMENTED** [review]
Request for membership

**Comment by morey-tech** [comment]
The Scalability SIG was officially approved during [the maintainer's meeting today](https://docs.google.com/document/d/1C71GA0XCxsyGP7Q4SYveHak5ZZcJOuaOiDCt_xHG10U/edit#heading=h.j7uq7oq0fxrt). This PR is ready to be merged.

**Review by hblixt — APPROVED** [review]
Approved as per the maintainer meeting 5/2.
Looking forward to lots of valuable content from this SIG!

**Pull request merged**
